Add: blog post on pycon us 2024 (#398)
pyOpenSci/pyopensci.github.io: _posts/2024-05-23-pyos-pyconus-2024-recap.md

---
layout: single
title: "pyOpenSci at PyCon US 2024 - Python packaging and community"
excerpt: "Learn about  pyOpenSci's experience at PyCon US 2024, how pyOpenSci is making Python packaging more accessible and beginner friendly and how the Python community is helping us get there."
author: "Leah Wasser"

> permalink: /blog/recap-pyos-pyconus-2024.html

Author: better url?

Author: ok great. i've read so many different things about url's and SEO for a while it was like long ones with lots of words and then it was shorter and concise. and now with AI... i have no idea.  😆

header:
  overlay_image: images/blog/2024/may/pycon-us-2024.png
  overlay_filter: 0.6
categories:
  - blog-post
  - community
classes: wide
toc: true
comments: true
---

<!-- TODOs:
* embed video when live
* add graphics from social that jesse sent
* fix captions
* add stravalib note at the end of my talk section -->

<div class="notice" markdown = "1">

> ## TL&DR

Reviewer: I've usually seen this as either:
* TLDR
* TL;DR

Author: Fixed!

> * pyOpenSci has a strong presence at pyCon US this year. I hope this continues for years to come! We held an open space, helped run the maintainers summit (lead by Inessa Pawson for 5 years and counting), gave a talk and ran a 1 day sprint where over 16 people contributed to our efforts.

Reviewer: ```suggestion
* pyOpenSci had a strong presence at [PyCon US](https://us.pycon.org/2024/) this year. I hope this continues for years to come! We held an open space, helped run the [Maintainers Summit](https://us.pycon.org/2024/events/maintainers-summit/) (lead by [Inessa Pawson](https://github.com/InessaPawson) for 5 years and counting), gave [a talk on Python packaging](https://us.pycon.org/2024/schedule/presentation/34/) and ran a 1 day sprint where over 16 people contributed to our efforts.
```

> * pyOpenSci’s theme this year for PyConUS was people first. People first when trying to make technical concepts easier to understand, people first when trying to write good tutorials or documentation and people first when you are trying to solve the world’s hardest problems.

Reviewer: ```suggestion
* pyOpenSci’s theme this year for PyConUS was people first: people first when trying to make technical concepts easier to understand, people first when trying to write good tutorials or documentation and people first when you are trying to solve the world’s hardest problems.
```

> * Giving a talk on packaging at pyConUS triggered every ounce of the imposter in me. But in the end it was a rewarding experience. Having friends in the audience made a world of difference. It was calming to focus on people who I know support both me and this vibrant organization. Friends really should never let friends … package … or use Python … or do anything technical…  alone.

Reviewer: ```suggestion
* Giving a talk on packaging at pyConUS triggered every ounce of the imposter in me. But in the end it was a rewarding experience. Having friends in the audience made a world of difference. It was calming to focus on people who I know support both me and this vibrant organization. Friends really should never let friends…package…or use Python…or do anything technical…alone.
```

</div>

> ## Another year, another pyCon

Reviewer: ```suggestion
## Another year, another incredible PyCon
```

> Wow! I wasn’t sure it would be possible to top last year’s PyCon US 2023 experience

Reviewer: ```suggestion
Wow! I wasn’t sure it would be possible to top [last year’s PyCon US 2023 experience](https://www.pyopensci.org/blog/pycon-2023-packaging-presentation-sprints-leah-wasser.html)
```

> in Salt Lake City, but this year's [PyConUS 2024](https://us.pycon.org/2024/) *was even better*. People are learning about pyOpenSci. The community is excited to support our mission to **help

Reviewer: ```suggestion
in Salt Lake City, but this year's [PyConUS 2024](https://us.pycon.org/2024/) *was even better*. People are learning about [pyOpenSci](https://www.pyopensci.org/). The community is excited to support our mission to **help
```

scientists create better software and make their science more open and
reproducible** so they can tackle the world's greatest challenges.

> PyConUS, run by the [Python Software

Reviewer: ```suggestion
PyCon US, run by the [Python Software
```

> Foundation,](https://www.python.org/psf-landing/) is one of the biggest Python meetings in the world with a record 2,700 registrations this year.

Reviewer: ```suggestion
Foundation,](https://www.python.org/psf-landing/) is one of the biggest Python meetings in the world, with a record 2,700 registrations this year.
```

{: .notice }

I also knew that I'd get to see a bunch of the friends who I met last year. I was returning to
this inclusive community, filled with Pythonistas like me
who care, who want to learn, and most importantly who want to help each other.

> And... Thanks to our incredible community manager,

Reviewer: ```suggestion
And...thanks to our incredible Community Manager,
```

[Jesse (@kiersi on GitHub)](https://github.com/kiersi), we now have

> a [pyOpenSci](https://www.pyopensci.org) visual brand -- and stickers galore.

Reviewer: YAY STICKERS!

Author: except you don't have any yet!! we need to fix that. and beautiful post cards too. but the stickers were popular. i think colored pencils for the sprint for the post cards are a must for scipy. just one set that people can use if they wish to take breaks from sprinting.

<figure>
  <img src="{{ site.baseurl }}/images/blog/2024/may/juno-stickers.png" alt="Image of a flat coated retriever chewing on a stick with a bunch of pyOpenSci stickers in front of her cut face. " style="width: 70%;" />
  <figcaption><i class="fa-solid fa-heart-pulse"></i> My rescue pup Juno prefers sticks, but also loves pyOpenSci <i class="fa-solid fa-heart-pulse"></i>.</figcaption>
</figure>

### So different from last year

My experience last year was a bit different. I walked into my first PyCon

> US - having no idea how I'd fit in at the meeting. I didn’t know

Reviewer: ```suggestion
US having no idea how I'd fit in at the meeting. I didn’t know
```

many people. I had never led a [mentored sprint](https://www.mentored-sprints.dev/).

> And I wasn't sure how my packaging talk would be received; you see, packaging is a notoriously thorny

Reviewer: ```suggestion
And I wasn't sure how my packaging talk, [Friends Don't Let Friends Package Alone](https://us.pycon.org/2024/schedule/presentation/34/) would be received; you see, packaging is a notoriously thorny
```

topic.

[Last year turned out pretty great - read more here.](pycon-2023-packaging-presentation-sprints-leah-wasser.html)

However, rather than feeling like an outsider, I met a lot of new people
who became new Python friends and colleagues. I learned about a lot of cool
projects and organizations and our sprints were awesome with many new contributions. Yaas! <i class="fa-solid fa-rocket"></i>

My takeaway: if you are considering going to PyCon but are worried about not knowing
people YOU SHOULD STILL GO! You can feel the true spirit of open source (and open science) at PyCon US. No judgement. All "levels" of Pythonistas welcome.

> ## My first main track PyCon US talk Friends don't let friends package alone

Reviewer: ```suggestion
## My first main track PyCon US talk: Friends don't let friends package alone
```

was unforgettable.

<figure>
  <img src="{{ site.baseurl }}/images/blog/2024/may/leah-talking-pyconus-2024.jpg" alt="Image of a woman on a big stage wearing a red tank top talking. On the side is a slide that says pyOpenSci - Save the Date. Open science fall festival september 28-29 2024" style="width: 70%;" />
  <figcaption><i class="fa-solid fa-heart-pulse"></i> Me talking on the stage! I was so nervous before but once I got up there, everything calmed in my mind.  <i class="fa-solid fa-heart-pulse"></i>.</figcaption>
</figure>

### A talk about making Python packaging more beginner friendly
My talk focused on how pyOpenSci helps beginners by breaking down
complex packaging concepts into simple, digestible pieces. I leaned into decades of
experience and study of teaching data science to various audiences, from big-ten university students
to students at under-resourced and under-served <a href="https://www.leahwasser.com/lessons-learned-working-with-tribal-hispanic-serving-schools-earth-data-science-corps" target="_blank">tribal colleges</a>. The essence of my talk was about the importance of

> community and collaboration in the thorny Python packaging ecosystem. And also why documentation should focus on fundamental concepts that users need to understand rather than tools.

Reviewer: ```suggestion
community and collaboration in the thorny Python packaging ecosystem, and also why documentation should focus on fundamental concepts that users need to understand, rather than tools.
```

from my talk:

* leverage community support,
* create early wins for users, focus on
fundamental concepts, and
* avoid “Too Many Options and Opinions” (TMO) to keep things simple
for beginners.

<figure>
  <img src="{{ site.baseurl }}/images/blog/2024/may/pyopensci-too-many-python-packaging-options-pycon-talk-2024.png
" alt="A cute image of a black flat coated retriever laying on the floor with a purple ice pack on it's head looking right at you." />
  <figcaption>Whether it's too many options or too many opinions, TMO will push beginner users away. Too many options create cognitive load that prevent users from having successful experiences.</figcaption>
</figure>

<!-- [I break this all down in more detail here if you want to read more.](#addlink) -->

Giving a talk at PyConUS experience reinforced my belief in the power of community to tackle complex
problems and support each other in our scientific journeys.

<!--
Or you can check out the talk itself on youtube below:

TODO: embed video - they are slowly being posted - 145 are in draft state....

2023 talk --
https://www.youtube.com/watch?v=Qxy7bxW72iA
-->

## The PyCon US maintainers Summit

> This year, I had the honor of helping Inessa Pawson and Kara Sowles organize the

Reviewer: ```suggestion
This year, I had the honor of helping I[nessa Pawson](https://github.com/InessaPawson) and [Kara Sowles](https://www.linkedin.com/in/karasowles/) organize the
```

[PyCon US Maintainers Summit 2024](https://us.pycon.org/2024/events/maintainers-summit/), an event that has grown from an experimental
“hatchery” to its 6th year at PyCon US. Last year, I had the [privilege of giving
a talk at the summit](https://www.youtube.com/watch?v=Qxy7bxW72iA). This year, I co-organized, learning the ropes of proposal
calls, speaker selection, and navigating a summit that filled a room of 80
people within a conference boasting 2,700 registrations. The prep leading up to
the summit was manageable, but the days before and the day of were quite the
adventure!

The summit featured amazing speakers, including:

> * Leonard Richardson, who shared

Reviewer: ```suggestion
* [Leonard Richardson](https://www.linkedin.com/in/leonardr/), who shared
```

insights on maintaining core tools like [Beautiful Soup](https://pypi.org/project/beautifulsoup4/) for 20 years and tips on
handling burnout.

> * [Jessica Tegner](https://github.com/JessicaTegner). Jessica became a maintainer of one of the top 1% of all downloaded packages on PyPI - [pypandoc](https://github.com/JessicaTegner/pypandoc). She talked about her experiences taking over a package with over 3 million monthly downloads. Oh and did I mention that Jessica does all this being visually challenged? The backgrounds (and dedication) of the maintainers in our open source ecosystem always blows my mind.

Reviewer: ```suggestion
* [Jessica Tegner](https://github.com/JessicaTegner). Jessica became a maintainer of one of the top 1% of all downloaded packages on PyPI - [pypandoc](https://github.com/JessicaTegner/pypandoc). She talked about her experiences taking over a package with over 3 million monthly downloads. Oh and did I mention that Jessica does all this while being visually challenged? The backgrounds (and dedication) of the maintainers in our open source ecosystem always blows my mind.
```

> Working alongside Inessa Pawson, Kara Sowles and Chris Rose who are all also rockstars in the open source world was a fantastic

Reviewer: ```suggestion
Working alongside Inessa Pawson, Kara Sowles and [Chris Rose](https://www.linkedin.com/in/chrisroseatcat/), who are all also rockstars in the open source world was a fantastic
```

experience. Despite some logistical challenges, like the summit filling up
quickly and a waitlist, the event was a success. Next year (if we are invited back), will be even better as we try to open the doors to include aspiring maintainers and contributors!

{% include video id="L-Ok_89QJOM" provider="youtube" %}

Also if you were wondering, yes that Monstera Deliciosa (plant) in the background is real! AND YES it is ginormous! <i class="fa-solid fa-leaf"></i>

## Our second pyOpenSci sprint

> This year we had a tremendous turnout of over 20 people from several countries for our 1-day pyConUS sprint. At last count this resulted in about [30 issues and pull requests](https://github.com/orgs/pyOpenSci/projects/12/views/1) where volunteers who were there to support scientists, helped us fix issues on our website, in our tutorials and our packaging and peer review guidebooks. So many long-standing issues and bugs were fixed thanks to this wonderful Python community.

Reviewer: ```suggestion
This year we had a tremendous turnout of over 20 people from several countries for our 1-day pyConUS sprint. At last count this resulted in about [30 issues and pull requests](https://github.com/orgs/pyOpenSci/projects/12/views/1) where volunteers who were there to support scientists, helped us fix issues on our website, in our tutorials and our [packaging](https://www.pyopensci.org/python-package-guide/) and [peer review](https://www.pyopensci.org/software-peer-review/index.html) guidebooks. So many long-standing issues and bugs were fixed thanks to this wonderful Python community.
```

<figure class="half">
  <img src="{{ site.baseurl }}/images/blog/2024/may/pycon-us-24-pyopensci-sprints-juno.png" alt="Image of the puycon us sprint board where projects signed up for different rooms." />
  <img src="{{ site.baseurl }}/images/blog/2024/may/pycon-24-sprints-pyopensci.png" alt="Image of the puycon us sprint board where projects signed up for different rooms." />

>   <figcaption>The pyConUS sprint sign-up board had lots of projects. Because the rooms are large projects tend to share spaces. We ended up in the packaging room which was great as it allowed us to do some more difficult work around Python packages with C extensions!</figcaption>

Reviewer: ```suggestion
  <figcaption>The pyConUS sprint sign-up board had lots of projects. Because the rooms are large, projects tend to share spaces. We ended up in the packaging room which was great as it allowed us to do some more difficult work around Python packages with C extensions!</figcaption>
```

</figure>

If you haven’t been to a sprint before, it’s an experience that every open source enthusiast should have. Sprints are where a bunch of people come together to work on a project. If you are running sprints that support software development in a professional environment (i.e. at a company) this might mean a team of people working together on releasing a new software feature. But for the open source community, sprints can also mean volunteers coming to a space to help maintainers work on various parts of a project that they care about - like pyOpenSci!

> pyOpenSci supports other people's software through it's [open community-lead peer review process ](https://www.pyopensci.org/about-peer-review/index.html) and it's online, free [packaging resources](https://www.pyopensci.org/python-package-guide/). But it also has it's own software too. We have tools that help us keep track of our review process, and volunteer contributions so we can acknowledge everyone for their effort.

Reviewer: ```suggestion
pyOpenSci supports other people's software through it's [open community-lead peer review process ](https://www.pyopensci.org/about-peer-review/index.html) and it's online, free [packaging resources](https://www.pyopensci.org/python-package-guide/). But it also has it's own software too. We have tools that help us keep track of our review process and volunteer contributions so we can acknowledge everyone for their effort.
```

  <figcaption>We had people working at multiple tables on pyOpenSci issues. A handful of people worked through our Create Your First Python Package tutorials and provided input and found issues! Included in this image is Jeremiah Paige who is making our packaging guide so much better by adding actual package examples (including on that has C extensions!) using Hatch! </figcaption>
</figure>

### How to run a sprint

I've learned a lot about running sprints over the past 2 years. At pyOpenSci, we have a [help-wanted project board](https://github.com/orgs/pyOpenSci/projects/3) where we add issues and things that we could use help with across our organization. Some of the tasks that we need help with are beginner friendly. For example it's useful for beginners to test drive our [online packaging tutorials](https://www.pyopensci.org/python-package-guide/tutorials/intro.html) and then report bugs as GitHub issues. Your feedback might be errors in a lesson, typos or things that were confusing / didn’t work.

> Some of our help-wanted issues are more technical too. pyOpenSci is a non-profit organization whose heart is a community of volunteers. We have infrastructure that any other open source project would have and as such we need technical help working on CI (continuous integration) workflows, guidebook infrastructure, our website and more.

Reviewer: ```suggestion
Some of our help-wanted issues are more technical too. pyOpenSci is a non-profit organization whose heart is a community of volunteers. We have infrastructure that any other open source project would have, and as such we need technical help working on CI (continuous integration) workflows, guidebook infrastructure, our website and more.
```
